### PR TITLE
feat: add WithCommandsFilter example (#126)

### PR DESCRIPTION
**Pull request by goravel-coder**
## Summary
- Add `WithCommandsFilter` builder example to scope registered Artisan commands by environment
- Add integration tests verifying filter returns nil in local env and built-in commands survive
- Add production env test using config override + `Restart()` to verify filter activation

## Why
This adds an example demonstrating the `WithCommandsFilter` builder method introduced in goravel/framework#1500. The filter callback returns a positive list of command signatures to keep in production (using exact and glob matching), and returns nil to keep all commands in dev/local environments.

```go
func CommandsFilter() []string {
	if facades.Config().GetString("app.env") == "production" {
		return []string{
			"up",
			"down",
			"make:*",
			"vendor:publish",
		}
	}

	return nil
}
```

The filter is wired into the builder chain in `bootstrap/app.go`:

```go
return foundation.Setup().
	// ...
	WithCommands(Commands).
	WithCommandsFilter(CommandsFilter).
	// ...
```

### CODE REVIEW

File: tests/feature/console_test.go [hunk 1]
```diff
@@ -148,3 +152,33 @@ func (s *ConsoleTestSuite) TestRunShutdownable() {
...
+	s.Require().NoError(err)
+	defer func() {
+		s.NoError(scope.Restore())
+	}()
+
+	filter := bootstrap.CommandsFilter()
+	s.NotNil(filter, "CommandsFilter should return non-nil in production")
+	s.Equal([]string{"up", "down", "make:*", "vendor:publish"}, filter)
+
+	s.NoError(facades.Artisan().Call("up"), "up should survive production filter")
```
Comment (hwbrzzl) [inline tests/feature/console_test.go:182]: Add an assertion that calling a custom command that is saved in `app/console/commands` to confirm if it's excluded in prod.

File: tests/feature/console_test.go [hunk 2]
```diff
@@ -148,3 +152,33 @@ func (s *ConsoleTestSuite) TestRunShutdownable() {
...
+	s.NoError(facades.Artisan().Call("about"))
+}
+
+func (s *ConsoleTestSuite) TestCommandsFilterKeepsGlobMatchesInLocalEnv() {
+	s.NoError(facades.Artisan().Call("make:command TestGlobFilterCmd"))
+	s.True(file.Contains(path.Bootstrap("commands.go"), "&commands.TestGlobFilterCmd{},"))
+}
+
+func (s *ConsoleTestSuite) TestCommandsFilterInProductionEnv() {
+	scope, err := telemetry.OverrideConfig(map[string]any{
```
Comment (hwbrzzl) [inline tests/feature/console_test.go]: Move `telemetry.OverrideConfig` to a general package instead of `telemetry`.
